client/js/board.js

```diff
@@ -109,6 +113,27 @@ export default class Board {
 		});
 	}
 
+	checkCollisionWithApples() {
+		this.snakes.forEach((snake, i) => {
+
+			let firstBodyPart = snake.bodyParts[0];
+
+			this.apples.forEach((apple, index) => {
+				if (firstBodyPart.x < apple.x + apple.radius * 2 &&
+	                firstBodyPart.x + firstBodyPart.width > apple.x &&
+	                firstBodyPart.y < apple.y + apple.radius * 2 &&
+	                firstBodyPart.height + firstBodyPart.y > apple.y) {
+
+					this.apples.splice(index, 1);
+					snake.addScore();
+					snake.addBodyPart();
+
+					this.emit('appleEaten', {'x':apple.x, 'y':apple.y});
```

Comment:
Cool.

Be aware of something that is not obvious. The [arrow function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) syntax is very recent and is the reason why this code works. If one of these arrow function was to be replaced with a normal function (`function(){}`), the `this` inside would be `undefined` and for example `this.apples` would throw an error (because one cannot get a property `apples` of the non-object undefined)